PyPDF2/_cmap.py

```diff
@@ -10,41 +10,95 @@
 # code freely inspired from @twiggy ; see #711
 def build_char_map(
     font_name: str, space_width: float, obj: DictionaryObject
-) -> Tuple[str, float, Dict[int, str], Dict]:
+) -> Tuple[
+    str, float, Dict[int, str], Dict
+]:  # font_type,space_width /2, encoding, cmap
     ft: DictionaryObject = obj["/Resources"]["/Font"][font_name]  # type: ignore
     font_type: str = cast(str, ft["/Subtype"])
 
     space_code = 32
     encoding, space_code = parse_encoding(ft, space_code)
     map_dict, space_code = parse_to_unicode(ft, space_code)
+    # encoding can be either a string for decode (on 1,2 or a variable number of bytes) of a char table (for 1 byte only for me)
+    # if empty string, it means it is than encoding field is not present and we have to select the good encoding from cmap input data
+    if encoding == "":
+        if map_dict[-1] == 1:
+            encoding = "charmap"
+        else:
+            encoding = "utf-16-be"
+    if font_name in _default_fonts_space_width:
+        # override space_width with new params
+        space_width = _default_fonts_space_width[font_name]
     sp_width = compute_space_width(ft, space_code, space_width)
 
     return (
         font_type,
         float(sp_width / 2),
-        dict(zip(range(256), encoding)),
+        encoding,
         # https://github.com/python/mypy/issues/4374
-        "".maketrans(map_dict),  # type: ignore
-    )
+        map_dict,  # type: ignore
+    )       # type: ignore
 
 
-def parse_encoding(ft: DictionaryObject, space_code: int) -> Tuple[List[str], int]:
-    encoding: List[str] = []
+_predefined_cmap: Dict[str, str] = {
+    "/Identity-H": "utf-16-be",
+    "/Identity-V": "utf-16-be",
+    "/GB-EUC-H": "gbk",  # TBC
+    "/GB-EUC-V": "gbk",  # TBC
+    "/GBpc-EUC-H": "gb2312",  # TBC
+    "/$WinREAgentGBpc-EUC-V": "gb2312",  # TBC
+}
+
+# manually extracted from http://mirrors.ctan.org/fonts/adobe/afm/Adobe-Core35_AFMs-229.tar.gz
```

Comment:
Wow, you keep impressing me!

Reply:
thanks for detecting the keyboard failure 😇